sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -803,7 +809,7 @@ int64_t CurlSession::OnRead(uint8_t* buffer, int64_t count, Context const& conte
 
   auto totalRead = int64_t();
   auto readRequestLength = this->m_isChunkedResponseType
-      ? std::min(this->m_chunkSize - this->m_sessionTotalRead, count)
+      ? (std::min)(this->m_chunkSize - this->m_sessionTotalRead, count)
```

Comment:
If we define `NOMINMAX` everytime before including `windows.h`. We don't need this extra `()`.

Reply:
Yep, understood. I am thinking we should add the extra `()` across all uses, anyway, for good measure.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/core/azure-core/test/ut/macro_guard.cpp

Reply:
I agree. However, would that catch all of them? It doesn't cover the usage in the *.cpp files.

Reply:
It's just personal preference. I don't like the macros, so I always try my best to avoid including them.

Reply:
In the case of this curl.cpp file, if I remove the `()` it gives me a compiler error for some reason.